vultr/resource_vultr_dns_domain.go

```diff
@@ -30,31 +28,53 @@ func resourceVultrDnsDomain() *schema.Resource {
 			},
 			"server_ip": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
+				ForceNew: true,
 			},
 		},
 	}
 }
 
 func resourceVultrDnsDomainCreate(d *schema.ResourceData, meta interface{}) error {
-
 	client := meta.(*Client).govultrClient()
 
 	domain := d.Get("domain").(string)
-	ip := d.Get("server_ip").(string)
+	ip, ipOk := d.GetOk("server_ip")
 
-	validIp := net.ParseIP(ip)
-	if validIp == nil {
-		return fmt.Errorf("The supplied IP address is invalid : %s", ip)
+	if ipOk {
+		validIp := net.ParseIP(ip.(string))
+		if validIp == nil {
+			return fmt.Errorf("The supplied IP address is invalid : %s", ip)
+		}
+	} else {
+		ip = "169.254.1.1"
 	}
 	log.Print("[INFO] Creating DNS domain")
 
-	err := client.DNSDomain.Create(context.Background(), domain, ip)
-
+	err := client.DNSDomain.Create(context.Background(), domain, ip.(string))
 	if err != nil {
 		return fmt.Errorf("Error while creating DNS domain : %s", err)
 	}
 
+	d.Set("server_ip", ip)
+	if ip == "169.254.1.1" {
+		d.Set("server_ip", nil)
+		log.Print("[INFO] DNS Domain : destroying default records")
+		records, err := client.DNSRecord.List(context.Background(), domain)
+		if err != nil {
+			return fmt.Errorf("error while retrieving records for delete : %s", err)
+		}
+
+		// clear out all records expect NS
+		for i := range records {
```

Comment:
nit: `for _, record` might be clearer since there will be less indexing below. Doesn't really matter either way though.